sdk/synapse/Azure.Analytics.Synapse.Spark/src/Customization/SparkSessionClient.cs

```diff
@@ -28,5 +31,285 @@ public SparkSessionClient(Uri endpoint, string sparkPoolName, TokenCredential cr
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary> Create new spark session. </summary>
+        /// <param name="sparkSessionOptions"> Livy compatible batch job request payload. </param>
+        /// <param name="detailed"> Optional query param specifying whether detailed response is returned beyond plain livy. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        internal virtual Response<SparkSession> CreateSparkSession(SparkSessionOptions sparkSessionOptions, bool? detailed = null, CancellationToken cancellationToken = default)
+        {
+            using var scope = _clientDiagnostics.CreateScope("SparkSessionClient.CreateSparkSession");
+            scope.Start();
+            try
+            {
+                return RestClient.CreateSparkSession(sparkSessionOptions, detailed, cancellationToken);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        public virtual async Task<SparkSessionOperation> StartGetSparkSessionAsync(int sessionId, bool? detailed = null, CancellationToken cancellationToken = default)
+        {
+            using var scope = _clientDiagnostics.CreateScope("SparkSessionClient.StartGetSparkSessionAsync");
+            scope.Start();
+            try
+            {
+                Response<SparkSession> session = await RestClient.GetSparkSessionAsync(sessionId, detailed, cancellationToken).ConfigureAwait(false);
+                return new SparkSessionOperation(this, this._clientDiagnostics, session);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        public virtual SparkSessionOperation StartGetSparkSession(int sessionId, bool? detailed = null, CancellationToken cancellationToken = default)
+        {
+            using var scope = _clientDiagnostics.CreateScope("SparkSessionClient.StartGetSparkSession");
+            scope.Start();
+            try
+            {
+                Response<SparkSession> session = RestClient.GetSparkSession(sessionId, detailed, cancellationToken);
+                return new SparkSessionOperation(this, _clientDiagnostics, session);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+       public virtual async Task<SparkSessionOperation> StartCreateSparkSessionAsync(SparkSessionOptions sparkSessionOptions, bool? detailed = null, CancellationToken cancellationToken = default)
+        {
+            using var scope = _clientDiagnostics.CreateScope("SparkSessionClient.StartCreateSparkSessionAsync");
```

Comment:
nit: consider using the `async` parameter pattern to avoid duplicating the body of the sync/async methods.
E.g. https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridPublisherClient.cs#L241-L257

Reply:
Not a bad point.